ui/pages/confirmations/components/alerts/alert-modal/alert-modal.tsx

```diff
@@ -0,0 +1,199 @@
+  alertKey,
+  onClose,
+  customButton,
+}: AlertModalProps) {
+  const t = useI18nContext();
+  const handleClose = useCallback(() => {
+    onClose();
+  }, [onClose]);
+  const { alerts, isAlertConfirmed, setAlertConfirmed } = useAlerts(ownerId);
+
+  const selectedAlert = alerts.find((alert) => alert.key === alertKey);
+
+  if (!selectedAlert) {
+    return null;
+  }
+  const isConfirmed = isAlertConfirmed(selectedAlert.key);
+  const severityStyle = getSeverityStyle(selectedAlert.severity);
+
+  return (
+    <Modal isOpen onClose={handleClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader onClose={handleClose}>
+          <Box
+            gap={3}
+            display={Display.Block}
+            alignItems={AlignItems.center}
+            textAlign={TextAlign.Center}
+          >
+            <Icon
+              name={
+                selectedAlert.severity === Severity.Info
+                  ? IconName.Info
+                  : IconName.Danger
+              }
+              size={IconSize.Xl}
+              color={severityStyle.icon}
+            />
+            <Text
+              variant={TextVariant.headingSm}
+              color={TextColor.inherit}
+              marginTop={3}
+            >
+              {selectedAlert.reason || t('alerts')}
+            </Text>
+          </Box>
+        </ModalHeader>
+        <ModalBody>
+          <Box
+            key={selectedAlert.key}
+            display={Display.InlineBlock}
+            padding={2}
+            width={BlockSize.Full}
+            backgroundColor={severityStyle.background}
+            gap={2}
+            borderRadius={BorderRadius.SM}
+          >
+            <Text variant={TextVariant.bodySm}>{selectedAlert.message}</Text>
+            {selectedAlert.alertDetails &&
+            selectedAlert.alertDetails?.length > 0 ? (
+              <Text variant={TextVariant.bodySmBold} marginTop={1}>
+                {t('alertModalDetails')}
+              </Text>
+            ) : null}
+
+            <Box
+              as="ul"
+              className={'alert-modal__alert-details'}
+              paddingLeft={6}
+            >
+              {selectedAlert.alertDetails?.map((detail, index) => (
+                <Box as="li" key={`${selectedAlert.key}-detail-${index}`}>
+                  <Text variant={TextVariant.bodySm}>{detail}</Text>
+                </Box>
+              ))}
+            </Box>
+          </Box>
+          <Box
+            display={Display.Flex}
+            padding={3}
+            width={BlockSize.Full}
+            gap={3}
+            backgroundColor={severityStyle.background}
+            marginTop={4}
+            borderRadius={BorderRadius.LG}
+          >
+            <Checkbox
+              label={t('alertModalAcknowledge')}
+              data-testid="alert-modal-acknowledge-checkbox"
+              isChecked={isConfirmed}
+              onClick={() => setAlertConfirmed(selectedAlert.key, !isConfirmed)}
```

Comment:
For easier maintenance, should we create a `handleAcknowledgeClick` function above?